latex-report/main.tex

```diff
@@ -641,14 +641,14 @@
                 return element
             count += 1
 
-    def __setitem__(self, index: int, val: typing.Union[int, None]):  # noqa: ANN204
+    def __setitem__(self, index: int, val: typing.Optional[int]=None):  # noqa: ANN204
```

Comment:
Nice, using the Optional type hint is better 👍🏻 
If you do end up editing this file again, add [spaces around the `=` for the default](https://www.python.org/dev/peps/pep-3107/#parameters) (Note that if there are no type-hints, no space is expected there)

Reply:
I'll make the change, but lets not print it out.